### DISCUSSION TITLE
Proxy or simulate gesture to sibling elements

### DISCUSSION THREAD

**Discussion by devinhalladay** [pmndrs/use-gesture#500] [Q&A]
Hi! Thanks for this awesome library, it seems way more user friendly than some of the alternatives!

I'm building an interaction — a PDF reader — that uses `usePinch()` to allow users to shrink their pages into a stack. The interaction would look something like this:
![CleanShot 2022-05-29 at 16 44 28](https://user-images.githubusercontent.com/1934813/170890856-200f3e11-d160-4b59-87e0-c678337f6380.gif)

```tsx
// Document.tsx houses the pinch logic
const bind = usePinch(
    ({
      origin: [ox, oy],
      first,
      movement: [ms],
      offset: [s, a],
      memo,
      event,
      args,
      ...rest
    }) => {
      return handlePinch({
        origin: [ox, oy],
        first,
        movement: [ms],
        offset: [s, a],
        memo,
        event,
        args,
        ...rest,
      });
    },
    {}
  );

  return (
    <div>
      {pages.map((_, i) => (
        <Page
          key={i}
          pagesRef={pagesRef}
          pageIndex={i}
          pages={pages}
          ref={(el) => (pagesRef.current[i] = el)}
          bind={bind}
        />
      ))}
    </div>
  );
```

```tsx
// Page.tsx renders a page and binds a gesture handler

const Page: React.FC<Props> = React.forwardRef(
  ({ pagesRef, pageIndex, pages, bind }, ref) => {
    const [style, api] = useSpring(() => ({
      translateY: 0,
    }));

    return (
      <animated.div
        key={pageIndex}
        ref={ref}
        {...bind(pageIndex, [style, api])}
        style={style}
      >
        ...
      </animated.div>
    );
  }
);
```

At the moment, this code obviously binds the pinch gesture to each element individually, so I can pinch on one and it will translate up.

What I'd like to achieve, as shown in the animation above, is to somehow share state between all `Page`s, such that pinching any given `Page` will trigger a translation on all other subsequent `Page`s.

How would I go about achieving this? Please let me know if I can make the question more clear!

**Comment by devinhalladay** [comment]
For further reference, my end goal is to recreate something similar to this interaction from the app Liquid Text:

![CleanShot 2022-05-29 at 16 53 24](https://user-images.githubusercontent.com/1934813/170891104-13cb80ee-dffa-4048-951d-e82bf4e7dbbd.gif)

**Reply by dbismut** [reply]
Maybe you could bind the pinch event to the whole window rather than one individual page?

**Reply by dbismut** [reply]
Otherwise I guess you could use a state management library, the problem being that you probably don't want to trigger renders on each frame? But maybe if you store the react spring animated value that contains the zoom level you might be able to pass it down to each page?

**Reply by dbismut** [reply]
Note that what I don't like with binding each page with its own handlers is that the pinch can only happen when both fingers are touching the same single page. Which imo is not what you want.
